### ISSUE TITLE
crop_like or resize_like

### ISSUE THREAD

**Issue by shahabty** [ClementPinard/SfmLearner-Pytorch#56]
Hello, 
I believe that the implementation of Dispnet is a little bit buggy. In Geonet and SFMLearner official Tensorflow repository, they use resize like which is the nearest neighbour interpolation. On the other hand, in this repo, the feature map is cropped. Could you please explain how this works?

**Comment by ClementPinard** [comment]
Define buggy :upside_down_face: 

I don't think nearest neighbour interpolation is the right choice here, because of problems with corner alignments. Features will then be dephased.

The interest of only cropping is to have the same behaviour between an image and a cropped version of this image.

for example if we run a convolution with stride 2 on a picture of size 416x128 , you will get a 208x64 image, and then with upconvolution of the same stride you get back a feature map of size 416x128 that you can concatenater with your initial image.

Now what happens if my image is of size 415x128 ? Because of how strided convolution works, the downsampled feature map will be the same size. and if you image this image as the cropped version of the 416x128 (so with one column less of pixels), a part from the last columns of features, the feature map will be exactly the same. Last column will be a bit different because padding makes the image assumed to have one more columns of 0-valued pixels

Now if you run the upconvolution on this feature map, you also get the same upsampled feature map as with the 416x128 example, and you need to make it 415x128 to concatenate the two tensors.

Do you see the problem here ? By resizing to 416x128, every column will be shifted : first column is not shift, 208th column gets shift of 0.5 pixels and last column gets shifted of 1 pixel.
As such, upsampled features and the image are now dephased while with just cropping you have avoided that.

Hope that was clear enough for you.
